tests/sgx_zerobase/host/host.cpp

```diff
@@ -44,55 +41,149 @@ int main(int argc, const char* argv[])
+     */
+    if (flags & OE_ENCLAVE_FLAG_SIMULATE)
+    {
+        fprintf(stdout, "Simulation mode does not support 0-base enclaves.\n");
+        return 0;
+    }
+
+    fprintf(
+        stdout,
+        "\nTest 2: Create 0-base enclave\n"
+        "Expected result : OE_OK\n");
+
     result = oe_create_sgx_zerobase_enclave(
         argv[1], OE_ENCLAVE_TYPE_SGX, flags, NULL, 0, &enclave);
     if (result != OE_OK)
     {
-        fprintf(stderr, "Could not create enclave, result=%d\n", result);
+        fprintf(stderr, "Could not create 0-base enclave, result=%d\n", result);
         return 1;
     }
-    char output[1024];
-    const char* source = "My First App\n";
+
+    fprintf(
+        stdout,
+        "\nTest 3: Test ecall, ocall on 0-base enclave\n"
+        "Expected result : OE_OK\n");
+
+    const char* input = "testing ecall\0";
     int res = -1;
-    OE_TEST(
-        secure_string_patching(
-            enclave, &res, source, output, OE_COUNTOF(output)) == OE_OK);
+
+    OE_TEST(test_ecall(enclave, &res, input) == OE_OK);
 
     if (res != 0)
     {
-        fprintf(stderr, "%s: enclave called failed\n", argv[0]);
-        exit(1);
+        fprintf(stderr, "[host]: ecall/ocall failed %d\n", res);
+        return 1;
     }
 
-    const char expect[] = "My First App\n"
-                          "Hello world from Enclave\n"
-                          "My First App\n"
-                          "Hello world from Host\n";
+    fprintf(
+        stdout,
+        "\nTest 4: Check host access between /proc/sys/vm/mmap_min_addr and "
+        "enclave image start address - [0x1000, 0x21000)\n"
+        "Expected result : success\n");
 
-    if (strcmp(output, expect) != 0)
+    uint64_t address = 0x15000;
+
+    if ((uint64_t)mmap(
+            (void*)address,
+            1,
+            PROT_READ | PROT_WRITE,
+            MAP_SHARED | MAP_ANONYMOUS,
+            -1,
+            0) != address)
     {
-        fprintf(stderr, "%s: returned string don't match\n", argv[0]);
+        fprintf(stderr, "host mmap at address %lx failed\n", address);
         return 1;
     }
 
+    /* Enclave memory access tests */
+    if (_is_misc_region_supported())
+    {
+        struct exceptions exception;
+
+        fprintf(
+            stdout,
+            "\nTest 5: Enclave memory access at 0x0\n"
+            "Expected result: page fault\n");
+
+        address = 0x0;
+
+        OE_TEST(test_enclave_memory_access(enclave, &res, address, &exception));
+        if (res != 0)
+        {
+            fprintf(stderr, "test_enclave_memory_access failed %d\n", res);
+            return 1;
+        }
+
+        fprintf(stdout, "address %lx, exception %d\n", address, exception.code);
+        if (!exception.exception)
+            return 1;
+
+        fprintf(
+            stdout,
+            "\nTest 6: memory access between enclave image start address "
+            "and the end of elrange - [0x21000, 0x40000)\n"
```

Comment:
How do we get the elrange?

Reply:
Added a comment to explain calculation of address elrange_end.